### DISCUSSION TITLE
SAXParser error on SpriteFramCache

### DISCUSSION THREAD

**Discussion by xCruss** [axmolengine/axmol#2127] [Q&A] [answered]
I tried following the tutorial in Using sprite sheet animations in Axmol Engine but I'm getting an error when I'm adding the sprites to the SpriteFrame Cache

`SpriteFrameCache::getInstance()->addSpriteFramesWithFile("cityscene.plist");`
`axmol: SAXParser: Error parsing xml: expected < at bplist00\322`

I have the same issue when I'm using other spritesheet not from the tutorial.

**Comment by rh101** [comment]
You need to provide one of the plist files that is triggering the issue, otherwise there is no way to tell what is going on.  Even better would be providing a minimal project that reproduces the issue.

Also, have you tested plist files (along with associated image files) from the cpp-tests project in your app to see if the same issue occurs?  If the test plist files work, then you can compare the contents of the them with the non-working files.

**Comment by xCruss** [comment]
[cityscene.zip](https://github.com/user-attachments/files/16843517/cityscene.zip)
Here are the files generated from texture packer that I added to the project.

**Reply by rh101** [reply]
I cannot reproduce the issue using the 2 files in that archive.  I added them to a project, and displayed a sprite from one of the frames in that sprite sheet:
```
auto spriteFrameCache = SpriteFrameCache::getInstance();
spriteFrameCache->addSpriteFramesWithFile("cityscene.plist");

auto sprite = Sprite::createWithSpriteFrameName("background.png");
sprite->setNormalizedPosition(Vec2::ANCHOR_MIDDLE);
sprite->setAnchorPoint(Vec2::ANCHOR_MIDDLE);
addChild(sprite);
```

The result:
![image](https://github.com/user-attachments/assets/69f08365-2ec5-443b-b1a6-bbe9650f8edf)

**Reply by rh101** [reply]
I can only suggest that you debug the code, and check the contents of the string being loaded from the plist file.  Ensure the content is actually correct.  If it is correct, then keep stepping through the code till the point of failure, and you should be able to see why it is happening.

**Reply by xCruss** [reply]
Thanks. But I'm still having the issue. I don't know why. I created a new project and just tried to load the spriteframes and didn't even use any image but the error is still there.

I just added this on the MainScene::init()

`SpriteFrameCache::getInstance()->addSpriteFramesWithFile("cityscene.plist");`

`SAXParser: Error parsing xml: expected < at bplist00\322hVframesXmetadata\331	


)2;DMV_^background.png_capguy/walk/0001.png_capguy/walk/0002.png_capguy/walk/0003.png_capguy/walk/0004.png_capguy/walk/0005.png_capguy/walk/0006.png_capguy/walk/0007.png_capguy/walk/0008.png\331Waliases\spriteOffsetZspriteSize_spriteSourceSize[textureRect^textureRotatedYtrianglesXverticesZverticesUV\240U{0,0}Z{1023,768}_{{1,1},{1023,768}}[1 2 3 0 1 3_1023 768 0 768 0 0 1023 0_1024 769 1 769 1 1 1024 1\331 !"#$&'(\240Y{0.5,0.5}X{70,129}X{75,130}_{{1026,138},{70,129}}_a13 0 3 13 3 12 8 11 12 13 14 0 8 10 11 10 8 9 4 6 7 4 5 6 4 7 8 12 3 4 12 4 8 0 1 2 0 2 3 0 14 15__52 7 66 1 72 1 45 60 54 97 73 113 68 118 48 129 36 104 27 129 15 129 3 117 20 94 14 6 40 0 43 0_\2171075 145 1089 139 1095 139 1068 198 1077 235 1096 251 1091 256 1071 267 1059 242 1050 267 1038 267 1026 255 1043 232 1037 144 1063 138 1066 138\331*+,#-/01\240X{-2.5,0}X{70,128}_{{1026,269},{70,128}}_Y0 13 14 11 3 6 7 10 11 7 8 9 7 9 10 6 7 11 11 12 3 6 4 5 6 3 4 0 3 12 0 1 2 0 2 3 0 12 13_Z49 9 63 2 68 2 46 68 70 125 43 125 40 104 16 112 12 129 0 117 0 101 22 88 15 33 18 12 37 1_\2061075 277 1089 270 1094 270 1072 336 1096 393 1069 393 1066 372 1042 380 1038 397 1026 385 1026 369 1048 356 1041 301 1044 280 1063 269\331345#689:\240W{2,2.5}X{65,125}_{{1026,542},{65,125}}_P0 12 13 1 12 0 1 10 11 6 8 9 6 7 8 3 4 5 3 5 6 3 6 9 2 3 9 10 1 2 10 2 9 1 11 12_T72 7 58 32 46 69 50 118 67 125 33 125 31 106 18 125 7 97 28 90 19 73 19 11 30 0 72 0_}1091 549 1077 574 1065 611 1069 660 1086 667 1052 667 1050 648 1037 667 1026 639 1047 632 1038 615 1038 553 1049 542 1091 542\331<=>#?ABC\240Y{7.5,0.5}X{58,129}_{{1078,415},{58,129}}_)0 7 8 0 6 7 1 5 6 2 3 4 1 2 4 1 4 5 0 1 6_474 9 46 56 52 129 25 122 25 91 16 73 16 35 25 0 74 0_P1136 424 1108 471 1114 544 1087 537 1087 506 1078 488 1078 450 1087 415 1136 415\331EFG#HJKL\240W{4,0.5}X{61,129}_{{1026,408},{61,129}}_G10 0 3 10 3 9 9 3 4 8 9 5 5 6 7 8 5 7 9 4 5 10 11 0 0 1 2 0 2 3 0 11 12_K52 7 66 1 72 1 46 57 56 89 46 103 66 118 41 129 21 119 11 68 18 7 40 0 43 0_t1067 415 1081 409 1087 409 1061 465 1071 497 1061 511 1081 526 1056 537 1036 527 1026 476 1033 415 1055 408 1058 408\331NOP#QSTU\240W{1.5,0}X{58,128}_{{1083,666},{58,128}}_X0 13 14 0 12 13 3 5 10 9 5 6 9 10 5 6 7 8 9 6 8 3 10 11 3 4 5 0 3 11 0 1 2 0 2 3 0 11 12_Y49 9 63 2 68 2 48 64 62 94 45 96 45 122 59 129 25 129 11 115 10 67 24 52 15 38 17 11 37 1_\2061122 674 1136 667 1141 667 1121 729 1135 759 1118 761 1118 787 1132 794 1098 794 1084 780 1083 732 1097 717 1088 703 1090 676 1110 666\331WXY#Z\]^\240U{4,1}X{61,128}_{{1026,669},{61,128}}_60 9 10 1 8 9 4 5 6 4 6 7 2 3 4 2 4 7 1 2 7 1 7 8 0 1 9_@72 7 46 58 49 69 65 86 39 112 46 128 20 128 11 67 21 7 42 0 72 0_b1087 676 1061 727 1064 738 1080 755 1054 781 1061 797 1035 797 1026 736 1036 676 1057 669 1087 669\331`ab#cefg\240U{2,0}X{69,130}_{{1026,1},{69,130}}_/0 8 9 1 7 8 1 6 7 6 2 3 6 3 4 6 1 2 6 4 5 0 1 8_:74 8 46 57 56 81 46 130 27 130 5 124 21 87 15 69 20 0 74 0_O1095 9 1067 58 1077 82 1067 131 1048 131 1026 125 1042 88 1036 70 1041 1 1095 1\327ijklmnopqstusVformat[pixelFormat_premultiplyAlpha_realTextureFileNameTsize[smartupdate_textureFileNameXRGBA8888]cityscene.pngZ{1142,798}_~$TexturePacker:SmartUpdate:580712a7766301cea76ad80ae75898cf:901a8c5b6c5ac754ede4d47255356dbc:781573cbd56cf75c56b063c5572d5827$`

**Reply by halx99** [reply]
The SAXParser not support parse binary plist file

**Reply by halx99** [reply]
> Thanks. But I'm still having the issue. I don't know why. I created a new project and just tried to load the spriteframes and didn't even use any image but the error is still there.
> 
> I just added this on the MainScene::init()
> 
> `SpriteFrameCache::getInstance()->addSpriteFramesWithFile("cityscene.plist");`
> 
> `SAXParser: Error parsing xml: expected < at bplist00\322���hVframesXmetadata\331����� �
> 
> �)2;DMV_^background.png_��capguy/walk/0001.png_��capguy/walk/0002.png_��capguy/walk/0003.png_��capguy/walk/0004.png_��capguy/walk/0005.png_��capguy/walk/0006.png_��capguy/walk/0007.png_��capguy/walk/0008.png\331������������������Waliases\spriteOffsetZspriteSize_��spriteSourceSize[textureRect^textureRotatedYtrianglesXverticesZverticesUV\240U{0,0}Z{1023,768}_��{{1,1},{1023,768}}�[1 2 3 0 1 3_��1023 768 0 768 0 0 1023 0_��1024 769 1 769 1 1 1024 1\331��������� !"#$�&'(\240Y{0.5,0.5}X{70,129}X{75,130}_��{{1026,138},{70,129}}�_�a13 0 3 13 3 12 8 11 12 13 14 0 8 10 11 10 8 9 4 6 7 4 5 6 4 7 8 12 3 4 12 4 8 0 1 2 0 2 3 0 14 15_�_52 7 66 1 72 1 45 60 54 97 73 113 68 118 48 129 36 104 27 129 15 129 3 117 20 94 14 6 40 0 43 0_�\2171075 145 1089 139 1095 139 1068 198 1077 235 1096 251 1091 256 1071 267 1059 242 1050 267 1038 267 1026 255 1043 232 1037 144 1063 138 1066 138\331���������*+,#-�/01\240X{-2.5,0}X{70,128}_��{{1026,269},{70,128}}�_�Y0 13 14 11 3 6 7 10 11 7 8 9 7 9 10 6 7 11 11 12 3 6 4 5 6 3 4 0 3 12 0 1 2 0 2 3 0 12 13_�Z49 9 63 2 68 2 46 68 70 125 43 125 40 104 16 112 12 129 0 117 0 101 22 88 15 33 18 12 37 1_�\2061075 277 1089 270 1094 270 1072 336 1096 393 1069 393 1066 372 1042 380 1038 397 1026 385 1026 369 1048 356 1041 301 1044 280 1063 269\331���������345#6�89:\240W{2,2.5}X{65,125}_��{{1026,542},{65,125}}�_�P0 12 13 1 12 0 1 10 11 6 8 9 6 7 8 3 4 5 3 5 6 3 6 9 2 3 9 10 1 2 10 2 9 1 11 12_�T72 7 58 32 46 69 50 118 67 125 33 125 31 106 18 125 7 97 28 90 19 73 19 11 30 0 72 0_�}1091 549 1077 574 1065 611 1069 660 1086 667 1052 667 1050 648 1037 667 1026 639 1047 632 1038 615 1038 553 1049 542 1091 542\331���������<=>#?�ABC\240Y{7.5,0.5}X{58,129}_��{{1078,415},{58,129}}�_�)0 7 8 0 6 7 1 5 6 2 3 4 1 2 4 1 4 5 0 1 6_�474 9 46 56 52 129 25 122 25 91 16 73 16 35 25 0 74 0_�P1136 424 1108 471 1114 544 1087 537 1087 506 1078 488 1078 450 1087 415 1136 415\331���������EFG#H�JKL\240W{4,0.5}X{61,129}_��{{1026,408},{61,129}}�_�G10 0 3 10 3 9 9 3 4 8 9 5 5 6 7 8 5 7 9 4 5 10 11 0 0 1 2 0 2 3 0 11 12_�K52 7 66 1 72 1 46 57 56 89 46 103 66 118 41 129 21 119 11 68 18 7 40 0 43 0_�t1067 415 1081 409 1087 409 1061 465 1071 497 1061 511 1081 526 1056 537 1036 527 1026 476 1033 415 1055 408 1058 408\331���������NOP#Q�STU\240W{1.5,0}X{58,128}_��{{1083,666},{58,128}}�_�X0 13 14 0 12 13 3 5 10 9 5 6 9 10 5 6 7 8 9 6 8 3 10 11 3 4 5 0 3 11 0 1 2 0 2 3 0 11 12_�Y49 9 63 2 68 2 48 64 62 94 45 96 45 122 59 129 25 129 11 115 10 67 24 52 15 38 17 11 37 1_�\2061122 674 1136 667 1141 667 1121 729 1135 759 1118 761 1118 787 1132 794 1098 794 1084 780 1083 732 1097 717 1088 703 1090 676 1110 666\331���������WXY#Z�]^\240U{4,1}X{61,128}_��{{1026,669},{61,128}}�_�60 9 10 1 8 9 4 5 6 4 6 7 2 3 4 2 4 7 1 2 7 1 7 8 0 1 9_�@72 7 46 58 49 69 65 86 39 112 46 128 20 128 11 67 21 7 42 0 72 0_�b1087 676 1061 727 1064 738 1080 755 1054 781 1061 797 1035 797 1026 736 1036 676 1057 669 1087 669\331���������`ab#c�efg\240U{2,0}X{69,130}_��{{1026,1},{69,130}}�_�/0 8 9 1 7 8 1 6 7 6 2 3 6 3 4 6 1 2 6 4 5 0 1 8_�:74 8 46 57 56 81 46 130 27 130 5 124 21 87 15 69 20 0 74 0_�O1095 9 1067 58 1077 82 1067 131 1048 131 1026 125 1042 88 1036 70 1041 1 1095 1\327ijklmnopq�stusVformat[pixelFormat_��premultiplyAlpha_��realTextureFileNameTsize[smartupdate_��textureFileName��XRGBA8888�]cityscene.pngZ{1142,798}_�~$TexturePacker:SmartUpdate:580712a7766301cea76ad80ae75898cf:901a8c5b6c5ac754ede4d47255356dbc:781573cbd56cf75c56b063c5572d5827$`

the error output indicate it's binary file

**Reply by xCruss** [reply]
But I'm using the same file I attached here which @rh101  were able to load. That's why I'm confused why it is not working on my end cause I can open the file as text and see that it is in the xml format. But when I try it in code I always get that error that says it is in binary.

**Reply by xCruss** [reply]
Ohhhhh. It was Xcode. There was a **Property List Output Encoding** in the Build Settings and it was set to binary by default. Thanks @halx99 and @rh101

**Comment by halx99** [comment, marked as answer]
the SAXParser not support parse binary plist file

**Reply by xCruss** [reply]
But it is not in binary plist file. It was generated from texturepacker and this is the content when opened as text:
<img width="731" alt="Screenshot 2024-09-04 at 1 45 14 PM" src="https://github.com/user-attachments/assets/f37e0dbc-0ae6-4f1b-844a-1ba8465a83b4">
